### PR TITLE
Add presentation media type

### PR DESCRIPTION
**Pull request by OR13**
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/pull/1032.html" title="Last updated on Feb 24, 2023, 5:06 PM UTC (23d4554)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/vc-data-model/1032/c344fa0...23d4554.html" title="Last updated on Feb 24, 2023, 5:06 PM UTC (23d4554)">Diff</a>

### CODE REVIEW

**Review by quartzjer — APPROVED** [review]

**Review by decentralgabe — APPROVED** [review]

**Review by mprorock — APPROVED** [review]

File: index.html
Comment (msporny) [inline index.html:5159]:
```suggestion
  <h2>The <code>application/presentation+ld+json</code> Media Type</h2>
```

I presume `application/verifiable-presentation+SOMETHING` is next? :)

File: index.html
Comment (msporny) [inline index.html:5161]:
```suggestion
    This specification registers the <code>application/presentation+ld+json</code> Media Type specifically for
```

IETF guidance is to not call them MIME (Multipurpose Internet Mail Extensions) Types anymore since the original concept grew past the goal of use in "internet mail" into other things like HTTP... From the IANA Media Types Registry -- "[[RFC2046](https://www.iana.org/go/rfc2046)] specifies that Media Types (formerly known as MIME types) and Media Subtypes will be assigned and listed by the IANA."

**Review by msporny — APPROVED** [review]
Tiny editorial nitpick, other than that, LGTM.

so. many. media types. :)

File: index.html
Comment (Sakurann) [inline index.html:5163]: Why use `verifiable presentations ` in the description of a media type for a `presentation`?

```suggestion
    conforming to the Presentations format.
```

File: index.html
Comment (TallTed) [inline index.html:5203]: It seems that @Sakurann's [question above](https://github.com/w3c/vc-data-model/pull/1032/files#r1103875272) applies here as well.  I'm also noticing an unexplained flipping between Presentation singular and Presentations plural, for which it seems we should pick one.
```suggestion
    Note that while the Presentations format uses JSON-LD
    conventions, there are a number of constraints and additional
    requirements for Presentation implementations that justify
```

File: index.html
Comment (OR13) [reply index.html:5163]: technical `verifiable presentations` NEVER required a `proof`.... I prefer to keep this sentence as is, to align with the way we registered `credential+ld+json`.

Comment (OR13) [reply index.html:5163]: however, we we feel very strongly about it, am willing to merge if we create an issue to make both registrations consistent.

File: index.html
Comment (OR13) [reply index.html:5203]: I'm holding on merging this, until folks way in on https://github.com/w3c/vc-data-model/pull/1032#discussion_r1106605757

File: index.html
Comment (msporny) [reply index.html:5163]: Suggest that we keep it the same as `application/credential+ld+json` for now. We should raise an issue to clarify the language to be consistent across both media types. I've raised issue #1052 to track @Sakurann's concern... which should allow us to merge this PR as long as @Sakurann is ok w/ that approach.

File: index.html
Comment (msporny) [reply index.html:5203]: I've raised issue #1052 to track @Sakurann and @TallTed's concerns... which should allow us to merge this PR as long as they are ok w/ that approach.

**Comment by msporny** [comment]
Merging this PR is currently waiting on @Sakurann and @TallTed to provide feedback on the path forward that was proposed in https://github.com/w3c/vc-data-model/pull/1032#discussion_r1106604869.

File: index.html
Comment (TallTed) [inline index.html:5110]:
```suggestion
    This specification registers the <code>application/credential+ld+json</code> Media Type specifically for
```

**Review by TallTed — CHANGES_REQUESTED** [review]
I know I've suggest this in (an)other PR(s). Not sure if it was on this document.

File: index.html
Comment (TallTed) [reply index.html:5163]: I'm fine with the format/media-type(s) always being labeled as Verifiable, as long as it's made clear everywhere that the Verifiability is not determined by conformance with the Media Type, but by the presence and provability of the optional proof(s), the lack of which deliver an UNverifiable Presentation/Credential.

**Review by iherman — CHANGES_REQUESTED** [review]
`application/presentation+ld+json` sounds awfully general to me. Following the logic of what the '+' sign means for media types, it would cover ***any JSON-LD file that has to do with anything "presentation"***, which is so general that it become meaningless.

The media type should be `application/presentation+credentials+ld+json` to really refer to what we want to express. And, I presume, following the same logic we would also have to have `application/presentation+verifiable±credentials+ld+json` media types.

Do we really want to go down that route?

**Comment by TallTed** [comment]
I *think* I would not be opposed to 
`application/credential+ld+json`
`application/verifiable+credential+ld+json`
`application/presentation+ld+json`
`application/verifiable+presentation+ld+json`

I think that these are self-documenting, within this group, and we should be able to write clearly enough about each to make them clear to people who have not participated in our discussions to date.

Note that each `verifiable+` type is a subtype of the preceding type, which offers no promise of verifiability but also offers no promise of UNverifiability.

File: index.html
Comment (selfissued) [reply index.html:5203]: I agree on using the singular "Presentation" rather than "Presentations".

**Review by selfissued — APPROVED** [review]
This seems like it's going in the right direction.

**Comment by OR13** [comment]
@iherman 

AFAIK, the only requirements for `presentation` are `@context` and `type: [ 'VerifiablePresentation' ] `

So I think your assessment is fairly accurate, however registering a media type for a defined data model is probably not the place to fix how useless / useful the data model is.

**Comment by iherman** [comment]
> @iherman
> 
> AFAIK, the only requirements for `presentation` are `@context` and `type: [ 'VerifiablePresentation' ] `
> 
> So I think your assessment is fairly accurate, however registering a media type for a defined data model is probably not the place to fix how useless / useful the data model is.
> 
> There are several APIs that already exchange this data model over http... and they would benefit from a way to be clear regarding it.

I do not see how this replies on my comment that `application/presentation+ld+json` is so general that it becomes useless. If the media type does not precisely define its content, why having it?

**Review by mprorock — APPROVED** [review]

**Review by mkhraisha — APPROVED** [review]

**Review by msporny — CHANGES_REQUESTED** [review]
Withdrawing support for this PR. The proliferation of media types does not seem to be helping us solve practical problems. We've run the experiment, and we should get back to focusing on media types that we know that people need for 90%+ of the use cases.

So, -1 on this PR... we don't need a media type for an unsecured presentation.

The group can probably agree to what is required for a VC (you have to secure the data model in some way to say it's a VC) and the same goes for a VP (it contains VCs, and it might also be secured).

The alternate chain of proposals is the following:

1. Rename `application/credential+ld+json` media type registration to `application/vc+ld+json`.
2. Register `application/vp+ld+json` as the media type for verifiable presentations. @brentzundel's argument is compelling, anything that contains a VC in it is a VP (whether or not the VP is secured or not, the mere existence of the VC in the VP makes it verifiable).
3. Use @mprorock's PR #1055 to define exactly what the expectations are around secured credentials and secured presentations, because at the end of the day, that's what's going to drive system behavior (not these unsecured things).
4. DO NOT register `application/credential+ld+json` and `application/presentation+ld+json` because they add little value.
5. Speak to the difference between credentials/presentations and verifiable credentials/verifiable presentations in the spec in a non-normative fashion.
6. Think about defining a `Credential` and `Presentation` RDF class after we work through 1-5 above.

Thoughts?

**Comment by TallTed** [comment]
> The alternate chain of proposals

I *think* I'm +1.  Magic 8 Ball says, "It is decidedly so."

**Comment by iherman** [comment]
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2023-03-14-vcwg-special#section2-1) on 2023-03-14

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 2.1. Add presentation media type (pr vc-data-model#1032)

_See github pull request [vc-data-model#1032](https://github.com/w3c/vc-data-model/pull/1032)._




**Manu Sporny:** That's three PRs that are knocked down and we can talk about PR #1032 next. It might be easier to address now.  

**Orie Steele:** We had previously merged the section that allowed us to talk about media types. I'd prefer to close PRs that I have opened that I authored and to make a single PR that implements the resolution that we took today. We can open other PRs to media types section. That's what I prefer to go forward. I don't want to retain all the arguments that are no longer relevant.  
… I'd like to open new PRs to address content in PRs I opened.  

> *Phillip Long:* +1 to Orie closing his pull requests.

**Orie Steele:** The same thing would go for presentations, we'd have the same conversation on vp+ld+json, I'd rather have that conversation on an issue. Manu you mentioned that PR 1032 — either we continue to discuss, or we can move to issue to discuss.  

**Brent Zundel:** To be clear, what's being suggested is closing 1014, and 1034, in favor of new PR that captures resolution we just made. Other PRs might be open for other language related to media type to be placed in the media types section.  

</details>

**Comment by OR13** [comment]
Closing this PR, we will attempt to define a media type for `vp+ld+json` in a new PR, after the editorial fixes for the section settle.

I will wait to open the new PR until https://github.com/w3c/vc-data-model/pull/1062 is merged.

**Pull request closed without merging**
